lib/UserAgentApplication.ts

```diff
@@ -183,24 +183,28 @@ namespace Msal {
         * @param _tokenReceivedCallback -  The function that will get the call back once this API is completed (either successfully or with a failure).
         * @param {boolean} validateAuthority -  boolean to turn authority validation on/off.
         */
-        constructor(clientId: string, authority: string, tokenReceivedCallback: tokenReceivedCallback, validateAuthority?: boolean) {
+        constructor(
+            clientId: string,
+            authority: string,
+            tokenReceivedCallback: tokenReceivedCallback,
+            validateAuthority?: boolean,
+            cacheLocation = 'sessionStorage'
+        ) {
+            this.cacheLocation = cacheLocation
             this.clientId = clientId;
-
             this.validateAuthority = validateAuthority === true;
-            this.authority = authority ? authority : "https://login.microsoftonline.com/common";
+            this.authority = authority || 'https://login.microsoftonline.com/common';
 
             if (tokenReceivedCallback) {
                 this._tokenReceivedCallback = tokenReceivedCallback;
             }
-
-            this.redirectUri = window.location.href.split("?")[0].split("#")[0];
+            this.redirectUri = window.location.href.split('?')[0].split('#')[0];
             this.postLogoutredirectUri = this.redirectUri;
             this._loginInProgress = false;
             this._acquireTokenInProgress = false;
             this._renewStates = [];
             this._activeRenewals = {};
-            this._cacheStorage = new Storage(this._cacheLocation); //cache keys msal
```

Comment:
@rohitnarula7176 
On line 193, the pre-existing setter for cache location is used. It was implemented but not used.

In the current code, this._cacheLocation gets defaulted to 'sessionStorage', so there was no way to set the cache location on the UserAgentApplication instance without mutating it.